sympy/logic/boolalg.py

```diff
@@ -1304,6 +1304,87 @@ def append_symbol(arg, symbols):
             for c in clauses]
 
 
+def term_to_integer(term):
+	"""
+	Constructs an integer from its base 2 digits.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import term_to_integer
+	>>> term_to_integer([1, 0, 1, 1])
+	11
+	>>> terms = [[0, 0, 0, 1], [0, 0, 1, 1],
+	...			[0, 1, 1, 1], [1, 0, 1, 1], [1, 1, 1, 1]]
+	>>> map(term_to_integer, terms)
+	[1, 3, 7, 11, 15]
+
+	"""
+
+	term = list(term)
+	return int(''.join(map(str, term)), 2)
+
+
+def integer_to_term(i, n_bits=0):
+	"""
+	Gives a list of the base 2 digits in the integer i.
+	If additional n_bits argument is passed,
+	integer_to_term() extends list of digits with zeros.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import integer_to_term
+	>>> integer_to_term(14)
+	[1, 1, 1, 0]
+	>>> integer_to_term(7, 5)
+	[0, 0, 1, 1, 1]
+	>>> truthtable = [1, 0, 0, 1, 1, 1, 0, 1]
+	>>> minterms = []
+	>>> n_bits = len(truthtable).bit_length() - 1
+	>>> for i, value in enumerate(truthtable):
+	...			if value:
+	...				minterms.append(integer_to_term(i, n_bits))
+	>>> minterms
+	[[0, 0, 0], [0, 1, 1], [1, 0, 0], [1, 0, 1], [1, 1, 1]]
+
+	"""
+	n = i.bit_length() if i else 1
+	prefix = [0] * (n_bits - n if n_bits else 0)
+	return prefix + map(int, list('{0:b}'.format(i)))
+
+
+def truth_table(expr, variables):
+	"""
+	Gives a list of truth values of the boolean expression.
```

Comment:
After reading your clarifications (which are much appreciated) I would suggest something like this:

```
Gives a list of truth values of the boolean expression in ``n`` variables where the i-th truth value is the output of the function when the input for each variable is ``integer_to_term(i, n)``.

EXAMPLES
========

The expression ``x >> y`` has 4 possible values corresponding to the
variables of x and y taking on values of 0 and 1:

>>> expr, vars = x >> y, [y, x]
>>> for i in range(2):
...     for j in range(2):
...         reps = list(zip(vars, (i, j)))
...         print reps, '->', int(bool(expr.subs(reps)))
...
[(y, 0), (x, 0)] -> 1
[(y, 0), (x, 1)] -> 0
[(y, 1), (x, 0)] -> 1
[(y, 1), (x, 1)] -> 1

This is represented more efficiently by the truth_table:

>>> truth_table(x >> y, [y, x])
[1, 0, 1, 1]

Note that the corresponding input values of variables can be deduced from the index of a
given output. For example, the False value at position 1 corresponds to an input
of y=0 and x=1:

>>> list(zip(vars, integer_to_term(1, len(vars))))
[(y, 0), (x, 1)]

The expression can also be given as a string,
...

```